packages/core/src/UserState.ts

```diff
@@ -360,545 +290,185 @@ export default class UserState extends Synchronizer {
         )
     }
 
-    /**
-     * Check if attester ID is valid
-     */
-    private _checkAttesterId = (attesterId: BigInt) => {
-        assert(
-            attesterId > BigInt(0),
-            `UserState: attesterId must be greater than zero`
-        )
-        assert(
-            attesterId < BigInt(2 ** this.settings.userStateTreeDepth),
-            `UserState: attesterId exceeds total number of attesters`
-        )
-    }
-
-    /**
-     * Generate the epoch key proof of the current user state.
-     * @param epochKeyNonce The nonce that is used in the epoch key proof.
-     * @returns The epoch key proof of type `EpochKeyProof`.
-     */
-    public genVerifyEpochKeyProof = async (
-        epochKeyNonce: number
-    ): Promise<EpochKeyProof> => {
-        this._checkEpkNonce(epochKeyNonce)
-        const epoch = await this.latestTransitionedEpoch()
-        const leafIndex = await this.latestGSTLeafIndex()
-        const epochKey = genEpochKey(
-            this.id.identityNullifier,
-            epoch,
-            epochKeyNonce,
-            this.settings.epochTreeDepth
+    public genAttestationProof = async (
```

Comment:
Ah it uses `getRepByEpochKey` which is in user state. We could put that in Synchronizer too, but i think this will change a lot when we do #142. Maybe we can do a better solution then.